edx_django_utils/monitoring/internal/middleware.py

```diff
@@ -400,18 +408,71 @@ def get_log_message_and_monitor_cookies(self, request):
-            if cookie_encryption_pub_key := getattr(settings, 'UNUSUAL_COOKIE_SAMPLING_PUBLIC_KEY', None):
-                headers_plain = dict(request.headers.items())
-                headers_enc = encrypt_for_log(json.dumps(headers_plain), cookie_encryption_pub_key)
-                log_message += f" All headers, with {corrupt_cookie_count} likely corruptions: {headers_enc}"
-
         return log_message
+
+    def log_corrupt_cookie_headers(self, request, corrupt_cookie_count):
+        """
+        Log all headers when corrupt cookies are detected (if settings permit).
+
+        This log data is encrypted using the log-sensitive utility.
+
+        - Logging requires that ``UNUSUAL_COOKIE_SAMPLING_PUBLIC_KEY`` is set.
+        - Output is split across multiple lines using ``UNUSUAL_COOKIE_SAMPLING_LOG_CHUNK``.
+        """
+        # Caller should ensure this, but check here anyway.
+        if corrupt_cookie_count < 1:
+            return
+
+        # .. setting_name: UNUSUAL_COOKIE_SAMPLING_PUBLIC_KEY
+        # .. setting_default: None
+        # .. setting_description: Use this public key to encrypt and log headers when there's
+        #   a corrupted-looking cookie in the request. See log_sensitive module for more detail
+        #   on the encryption. If no key is provided, this logging is skipped. Also see
+        #   ``UNUSUAL_COOKIE_SAMPLING_LOG_CHUNK``.
+        corrupt_cookie_log_pub_key = getattr(settings, 'UNUSUAL_COOKIE_SAMPLING_PUBLIC_KEY', None)
+
+        if not corrupt_cookie_log_pub_key:
+            return
+
+        # .. setting_name: UNUSUAL_COOKIE_SAMPLING_LOG_CHUNK
+        # .. setting_default: 9000
+        # .. setting_description: If necessary, logs data in chunks of this size, splitting across
+        #   multiple log messages. This should be set with your deployment's maximum log message
+        #   size in mind: Setting it too high may result in truncated messages, which will prevent
+        #   decryption (CryptoError).
+        chunk_size = getattr(settings, 'UNUSUAL_COOKIE_SAMPLING_LOG_CHUNK', 9000)
+
+        header_data = json.dumps(dict(request.headers.items()))
+        enc_output = encrypt_for_log(header_data, corrupt_cookie_log_pub_key)
+        msg = f"All headers for request with corrupted cookies (count={corrupt_cookie_count}): {enc_output}"
+
+        for piece in split_ascii_log_message(msg, chunk_size):
+            log.info(piece)
+
+
+def split_ascii_log_message(msg, chunk_size):
+    """
+    Generator that splits a message string into chunks of at most ``chunk_size`` characters.
+
+    Message must consist of single-byte (ASCII-range) characters, otherwise chunks will
+    not reliably be capped at the ``chunk_size``.
+
+    A small collation tag will be added to the end of each chunk, and it may not be possible
+    to reliably predict the length of a log message's prefix (in its final output format), so
+    the ``chunk_size`` should be set conservatively.
```

Comment:
Does conservatively mean 10, 100, 1000, etc.? Based on the default, for us it means 1000, so maybe you could provide 1000 characters as an example.

Reply:
Conservatively relative to absolute max log line size. If the max is 10,000, then chunk size should be maybe 9,000.

(This is different from the previous max cookie size to log, since this is _after_ expansion due to Base64.)